lib/wallet/src/Cardano/Wallet/Transaction.hs

```diff
@@ -360,6 +367,71 @@ data DelegationAction
 instance Buildable DelegationAction where
     build = genericF
 
+data DepositEvent = DepositTaken !Coin | DepositReturned !Coin
+    deriving (Eq, Generic, Show)
+
+data DepositEventParams = DepositEventParams
+    { protocolParameterDeposit :: !Coin
+    , txResolvedInputs :: ![Coin]
```

Comment:
@Unisay I wonder if there's much point to making this field (and others like it) strict, given that lists are lazily evaluated anyway? (Unless of course we make sure to force the contents of the list when we construct the record.)